trin-core/src/utp/stream.rs

```diff
@@ -103,30 +111,51 @@ struct DelayDifferenceSample {
 /// Represent overlay to uTP listener request. It is used as a way to communicate between the overlay protocol
 /// and uTP listener
 pub enum UtpListenerRequest {
-    /// Request to listen for Accept stream
-    AcceptStream(ConnId, Vec<Vec<u8>>),
-    /// Request to initialize uTP streram with remote node
-    Connect(ConnId, NodeId, oneshot::Sender<anyhow::Result<UtpSocket>>),
-    /// Request to listen for FindCOntent stream and send content data
-    FindContentData(ConnId, ByteList),
-    /// Request to listen for FindContent stream
-    FindContentStream(ConnId),
-    /// Request to listen for Offer stream
-    OfferStream(ConnId),
+    /// Request to initialize uTP stream with remote node
+    Connect(
+        ConnId,
+        NodeId,
+        ProtocolId,
+        UtpStreamId,
+        oneshot::Sender<anyhow::Result<UtpStream>>,
+    ),
+    /// Request to add uTP stream to the active connections
+    AddActiveConnection(Enr, ProtocolId, UtpStreamId, ConnId),
```

Comment:
I wonder if `InitiateConnection` would be a more descriptive name for this event.

The name `AddActiveConnection` is a little bit confusing to me because it seems like both `Connect` and `AddActiveConnection` events add an active connection to the `utp_connections`. In fact, `Connect` seems like it creates more of an "active" connection than `AddActiveConnection` does, because by the time a `Connect` event happens we necessarily have two participants in the stream, whereas an `AddActiveConnection` might only ever result in an inactive connection if the other peer never connects.

Reply:
You are right, the idea of `AddActiveConnection` is to create an uTP stream and start listening for incoming SYN packets.
`Connect` is creating an uTP stream and is sending SYN packet for a handshake with the remote peer.